bevy_lint/README.md

```diff
@@ -1,9 +1,151 @@
+
+You can install the toolchain with:
+
+```bash
+rustup toolchain install $TOOLCHAIN_VERSION \
+    --component rustc-dev \
+    --component llvm-tools-preview
+```
+
+For example, you would replace `$TOOLCHAIN_VERSION` with `nightly-2024-10-03` if you were installing `bevy_lint` 0.1.0. Please be aware that you must keep this toolchain installed for `bevy_lint` to function[^keep-toolchain-installed].
+
+[^keep-toolchain-installed]: `bevy_lint` imports internal `rustc` libraries in order to hook into the compiler process. These crates are stored in a [dynamic library](https://en.wikipedia.org/wiki/Dynamic_linker) that is installed with the `rustc-dev` component and loaded by `bevy_lint` at runtime. Uninstalling the nightly toolchain would remove this dynamic library, causing `bevy_lint` to fail.
+
+Once you have the toolchain installed, you can compile and install `bevy_lint` through `cargo`:
+
+```bash
+rustup run $TOOLCHAIN_VERSION cargo install \    
+    --git https://github.com/TheBevyFlock/bevy_cli.git \
+    --tag $TAG \
+    --locked \
+    bevy_lint
+```
+
+Make sure to replace `$TOOLCHAIN_VERSION` and `$TAG` in the above command. The tag for a specific release can be found in the [releases tab](https://github.com/TheBevyFlock/bevy_cli/releases). For example, the tag for 0.1.0 is `lint-v0.1.0`.
+
+## Usage
+
+`bevy_lint` has the same API as the `cargo check` command:
+
+```bash
+bevy_lint --help
+```
+
+If you have the Bevy CLI installed, the linter is also available through the `lint` subcommand:
+
+```bash
+bevy lint --help
+```
+
+<div class="rustdoc-alert rustdoc-alert-note">
+
+> **Note**
+>
+> `bevy_lint` checks your code with the nightly toolchain it was installed with, meaning you _do_ have access to unstable features when it is called. This is best used when [detecting `bevy_lint`](#detecting-bevy_lint).
+
+</div>
+
+### Detecting `bevy_lint`
+
+The linter passes `--cfg bevy_lint` when it checks your code, allowing you to detect it:
+
+```rust,ignore
+// Conditionally include this function only when `bevy_lint` is used.
+#[cfg(bevy_lint)]
+fn foo() {
+    // ...
+}
+
+// Conditionally add an attribute only when `bevy_lint` is used.
+#[cfg_attr(bevy_lint, ...)]
+struct Foo;
+```
+
+If you use this, you may also need to register `bevy_lint` as a valid `cfg` flag in your `Cargo.toml`:
+
+```toml
+[lints.rust]
+unexpected_cfg = { level = "warn", check-cfg = ["cfg(bevy_lint)"] }
+```
+
+### Configuring Lints
+
+If you wish to enable and disable certain lints, you must register `bevy` as a tool. Not doing so will cause `#[allow(bevy::lint_name)]` and related attributes to fail to compile.
+
+You can register a new tool using the `#![register_tool(...)]` attribute, which is [currently unstable](https://doc.rust-lang.org/nightly/unstable-book/language-features/register-tool.html). This isn't an issue if you [detect when `bevy_lint` is enabled](#detecting-bevy_lint), since it is guaranteed to check your code using nightly Rust.
```

Comment:
I feel like this paragraph belongs in the "detecting bevy_lint" section, under an "Alternatives" heading.

Reply:
I'm not sure I agree, since `#![register_tool(...)]` fulfills a different purpose than `--cfg bevy_lint`.

`--cfg bevy_lint` is so the code author can enable extra nightly features or disable portions of code when we know `bevy_lint` is being used to check the code.

`#![register_tool(bevy)]` is so `rustc` knows `bevy::` is a valid prefix in attributes, **even when `bevy_lint` is not used**. It's the same mechanism that lets you use `#[rustfmt::skip]` or `#[clippy::author]` without having a crate named `rustfmt` or `clippy` in the workspace. `#[rustfmt::skip]` is silently ignored by the compiler when `rustfmt` is not run, but this is only because it was previously registered as a tool.

I do agree that the commented paragraph is the odd-one-out in this section, though. It's more of a prerequisite to toggling lints than actually part of the process. I think I'm going to elaborate registering `bevy` as a tool into a separate section, which will be linked in here.

Reply:
Ok, I finished refactoring this in ef3ffee575c5a74e1665629562b2bfab8628c3d3. Tell me what you think!